### PR TITLE
Internal: upgrade to Flow 0.191.0

### PR DESCRIPTION
**Pull request by AlbertCarreras**
Internal: upgrade to Flow 0.191.0

https://github.com/facebook/flow/releases/tag/v0.191.0

### CODE REVIEW

File: packages/gestalt-datepicker/src/DateField.js [hunk 1]
```diff
@@ -6,40 +6,55 @@ import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
 import InternalDateField from './DateField/InternalDateField.js';
 
 // LocaleData type from https://github.com/date-fns/date-fns/blob/81ab18785146405ca2ae28710cdfbb13a294ec50/src/locale/af/index.js.flow
-// flowlint unclear-type:off
 // NOTE: DO NOT USE PER-LINE FLOW SUPPRESSIONS HERE
```
Comment (dangerismycat) [inline packages/gestalt-datepicker/src/DateField.js:9]: Screen Shot 2023-04-25 at 6 17 38 PM

we can't use per-line Flow suppressions here

File: packages/gestalt/src/Video.js [hunk 2]
```diff
@@ -626,7 +626,7 @@ export default class Video extends PureComponent<Props, State> {
   // Sent periodically to inform interested parties of progress downloading the media
   handleProgress: (event: SyntheticEvent<HTMLVideoElement>) => void = (event) => {
     const { onLoadedChange } = this.props;
-    const { buffered } = this.video || {};
+    const { buffered } = this.video || { buffered: undefined };
```
Comment (dangerismycat) [inline packages/gestalt/src/Video.js:629]: Consider instead
```js
const buffered = this.video?.buffered;
```

File: packages/gestalt/src/style.js [hunk 3]
```diff
@@ -59,21 +59,21 @@ export const mapClassName =
...
 export const toProps = ({ className, inlineStyle }: Style): ToPropsOutput => {
-  const props = {};
-
+  let classNameTemp;
   if (className.size > 0) {
     // Sorting here ensures that classNames are always stable, reducing diff
     // churn. Box usually has a small number of properties so it's not a perf
     // concern.
-    props.className = Array.from(className).sort().join(' ');
+    classNameTemp = Array.from(className).sort().join(' ');
```
Comment (dangerismycat) [inline packages/gestalt/src/style.js:70]: nit: `sortedClassNames`

File: packages/gestalt/src/style.js [hunk 4]
```diff
@@ -59,21 +59,21 @@ export const mapClassName =
...
   }
+  let styleTemp;
 
   if (Object.keys(inlineStyle).length > 0) {
-    props.style = inlineStyle;
+    styleTemp = inlineStyle;
```
Comment (dangerismycat) [inline packages/gestalt/src/style.js:75]: nit: `stylesObj`

File: playwright/masonry/load-more.spec.mjs [hunk 5]
```diff
@@ -65,7 +65,10 @@ test.describe('Masonry: scrolls', () => {
     // Scroll a few times to triggle multiple scrolls.
     await page.evaluate(() => {
       // eslint-disable-next-line playwright/no-conditional-in-test
-      const documentElement = document.documentElement || {};
+      const documentElement = document.documentElement || {
```
Comment (dangerismycat) [inline playwright/masonry/load-more.spec.mjs:68]: nit: consider defining the default value once, at the top of the file — something like `defaultDocumentElement`

File: playwright/masonry/optimized-remount.spec.mjs [hunk 6]
```diff
@@ -18,7 +18,10 @@ test.describe('Masonry: external cache', () => {
     // Scroll a couple of times.
     await page.evaluate(() => {
       // eslint-disable-next-line playwright/no-conditional-in-test
-      const documentElement = document.documentElement ?? {};
+      const documentElement = document.documentElement ?? {
```
Comment (dangerismycat) [inline playwright/masonry/optimized-remount.spec.mjs:21]: same nit here

File: packages/eslint-plugin-gestalt/src/helpers/styleValidators.js [hunk 7]
```diff
@@ -58,7 +58,8 @@ export function genBointLookup(
   start: number,
   end: number = 12,
 ): GenBointLookupType {
-  const lookupMap = {};
+  const lookupMap: {| [string | number]: string |} = {};
```
Comment (dangerismycat) [inline packages/eslint-plugin-gestalt/src/helpers/styleValidators.js:61]: nit: define this type once and use it in both places

File: docs/integration-test-helpers/masonry/MasonryContainer.js [hunk 8]
```diff
@@ -304,14 +366,16 @@ export default class MasonryContainer extends Component<Props, State> {
     }
 
     if (noScroll) {
-      dynamicGridProps.scrollContainer = () => undefined;
+      dynamicGridProps.scrollContainer = undefined;
```
Comment (dangerismycat) [inline docs/integration-test-helpers/masonry/MasonryContainer.js:369]: Why this change? This is a runtime change.

File: docs/integration-test-helpers/masonry/MasonryContainer.js [hunk 9]
```diff
@@ -304,14 +366,16 @@ export default class MasonryContainer extends Component<Props, State> {
...
     } else if (hasScrollContainer) {
-      dynamicGridProps.scrollContainer = () =>
-        typeof document === 'undefined'
-          ? undefined
-          : document.querySelector('[data-scroll-container]');
+      const query = document.querySelector('[data-scroll-container]');
+      if (typeof document === 'undefined' || query === null) {
```
Comment (dangerismycat) [inline docs/integration-test-helpers/masonry/MasonryContainer.js:372]: `query === null` is a runtime change — why is it needed?

File: docs/integration-test-helpers/masonry/MasonryContainer.js [hunk 10]
```diff
@@ -304,14 +366,16 @@ export default class MasonryContainer extends Component<Props, State> {
...
+        dynamicGridProps.scrollContainer = undefined;
```
Comment (dangerismycat) [inline docs/integration-test-helpers/masonry/MasonryContainer.js:373]: This is a runtime change

File: docs/integration-test-helpers/masonry/MasonryContainer.js [hunk 11]
```diff
@@ -304,14 +366,16 @@ export default class MasonryContainer extends Component<Props, State> {
...
+      } else if (query) {
+        dynamicGridProps.scrollContainer = () => query;
+      }
     } else {
-      dynamicGridProps.scrollContainer = () => (typeof window === 'undefined' ? undefined : window);
+      dynamicGridProps.scrollContainer = typeof window === 'undefined' ? undefined : () => window;
```
Comment (dangerismycat) [inline docs/integration-test-helpers/masonry/MasonryContainer.js:378]: This is a runtime change

File: docs/integration-test-helpers/masonry/MasonryContainer.js [hunk 12]
```diff
@@ -199,7 +199,7 @@ export default class MasonryContainer extends Component<Props, State> {
     this.forceUpdate();
   };
 
-  loadItems: ({| force: boolean, from?: number, name?: string |}) => void = ({
+  customLoadItems: ({| force: boolean, from?: number, name?: string |}) => void = ({
```
Comment (dangerismycat) [inline docs/integration-test-helpers/masonry/MasonryContainer.js:202]: Why this change?

File: docs/integration-test-helpers/masonry/MasonryContainer.js [hunk 13]
```diff
@@ -125,7 +125,7 @@ export default class MasonryContainer extends Component<Props, State> {
 
   handleAddItems: () => void = () => {
     const { items } = this.state;
-    this.loadItems({ name: 'Manual Fetch Pin', from: items.length, force: true });
+    this.customLoadItems?.({ name: 'Manual Fetch Pin', from: items.length, force: true });
```
Comment (dangerismycat) [inline docs/integration-test-helpers/masonry/MasonryContainer.js:128]: This is a method defined on the class, it will always be defined — no need for optional chaining

**Review by dangerismycat — CHANGES_REQUESTED** [review]
I'm curious about the runtime changes in MasonryContainer — what's motivating those?

File: packages/gestalt/src/style.js (hunk 3 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/style.js:70]: DONE

File: packages/gestalt/src/style.js (hunk 4 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/style.js:75]: DONE

File: playwright/masonry/load-more.spec.mjs (hunk 5 above)
Comment (AlbertCarreras) [reply playwright/masonry/load-more.spec.mjs:68]: DONE

File: playwright/masonry/optimized-remount.spec.mjs (hunk 6 above)
Comment (AlbertCarreras) [reply playwright/masonry/optimized-remount.spec.mjs:21]: DONE

File: packages/eslint-plugin-gestalt/src/helpers/styleValidators.js (hunk 7 above)
Comment (AlbertCarreras) [reply packages/eslint-plugin-gestalt/src/helpers/styleValidators.js:61]: DONE

File: docs/integration-test-helpers/masonry/MasonryContainer.js (hunk 8 above)
Comment (AlbertCarreras) [reply docs/integration-test-helpers/masonry/MasonryContainer.js:369]: @dangerismycat 

scrollContainer can't be `() => void`, it should be `() => HTMLElement`

But it can be optional / undefined.

Got a better approach for this?

Screenshot by Dropbox Capture

File: docs/integration-test-helpers/masonry/MasonryContainer.js (hunk 9 above)
Comment (AlbertCarreras) [reply docs/integration-test-helpers/masonry/MasonryContainer.js:372]: This is from previous attempts to make Flow be happy with querySelector being able to return null and         `dynamicGridProps.scrollContainer = () => query;` having to return HTMLElement

File: docs/integration-test-helpers/masonry/MasonryContainer.js (hunk 10 above)
Comment (AlbertCarreras) [reply docs/integration-test-helpers/masonry/MasonryContainer.js:373]: Same as  https://github.com/pinterest/gestalt/pull/2867#discussion_r1177215478

File: docs/integration-test-helpers/masonry/MasonryContainer.js (hunk 11 above)
Comment (AlbertCarreras) [reply docs/integration-test-helpers/masonry/MasonryContainer.js:378]: Same as https://github.com/pinterest/gestalt/pull/2867#discussion_r1177215478
technically test is incorrect as it cant  be `() => void`

File: docs/integration-test-helpers/masonry/MasonryContainer.js (hunk 12 above)
Comment (AlbertCarreras) [reply docs/integration-test-helpers/masonry/MasonryContainer.js:202]: Because loadItems doesn't have `name` nor `force` in the Flow type. 

Screenshot by Dropbox Capture

File: docs/integration-test-helpers/masonry/MasonryContainer.js (hunk 8 above)
Comment (dangerismycat) [reply docs/integration-test-helpers/masonry/MasonryContainer.js:369]: Ah ok, that makes sense. This (and similar changes) are fine, then. Thanks for the explanation 🙂

File: docs/integration-test-helpers/masonry/MasonryContainer.js (hunk 9 above)
Comment (dangerismycat) [reply docs/integration-test-helpers/masonry/MasonryContainer.js]: Makes sense!

**Review by dangerismycat — APPROVED** [review]
Looks good, thanks for the explanations!
